test/e2e/utils_test.go

```diff
@@ -658,3 +662,75 @@ func verifyWorkPropagationAndMarkAsApplied(memberClusterName, crpName string, re
 		}, eventuallyDuration, eventuallyInterval).Should(Succeed())
 	}
 }
+
+func buildTaints(memberClusterNames []string) []clusterv1beta1.Taint {
+	var labels map[string]string
+	taints := make([]clusterv1beta1.Taint, len(memberClusterNames))
+	for i, name := range memberClusterNames {
+		labels = labelsByClusterName[name]
```

Comment:
Also, I am not 100% sure about deriving taints/tolerations from the labelsByClusterName top-level map -> there's no guarantee that all clusters have the required labels.

Reply:
This implicit assumption can be troublesome if the global map is updated later.